debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java

```diff
@@ -127,18 +126,24 @@ public SqlServerStreamingChangeEventSource(SqlServerConnectorConfig connectorCon
             // otherwise we might skip an incomplete transaction after restart
             boolean shouldIncreaseFromLsn = offsetContext.isSnapshotCompleted();
 
-            if (offsetContext.getStreamingExecutionState() != null && offsetContext.getStreamingExecutionState().getTablesSlot() != null) {
+            boolean hasExecutionState = offsetContext.getStreamingExecutionState() != null;
+            boolean hasInitializedTableSlot = hasExecutionState
+                    && offsetContext.getStreamingExecutionState().getTablesSlot() != null
+                    && offsetContext.getStreamingExecutionState().getTablesSlot().length > 0;
+            if (hasExecutionState) {
                 schemaChangeCheckpoints = offsetContext.getStreamingExecutionState().getSchemaChangeCheckpoints();
-                tablesSlot = offsetContext.getStreamingExecutionState().getTablesSlot();
                 lastProcessedPositionOnStart = offsetContext.getStreamingExecutionState().getLastProcessedPositionOnStart();
                 lastProcessedEventSerialNoOnStart = offsetContext.getStreamingExecutionState().getLastProcessedEventSerialNoOnStart();
                 changesStoppedBeingMonotonic = offsetContext.getStreamingExecutionState().getChangesStoppedBeingMonotonic();
 
                 lastProcessedPosition = offsetContext.getStreamingExecutionState().getLastProcessedPosition();
                 shouldIncreaseFromLsn = offsetContext.getStreamingExecutionState().getShouldIncreaseFromLsn();
+
+                if (hasInitializedTableSlot) {
```

Comment:
This looks like a logical error. Why do we initialize `tablesSlot` if it already has been initialized? A similar block below has the inverse condition.